Contributing.md

```diff
@@ -30,10 +30,24 @@ This project uses clang-format (stable branch) to check for common style issues.
 
 In most cases, clang-format can and **should** be used to automatically reformat code and solve most formatting issues.
 
-- To run clang-format on all staged files:
+It can be downloaded as part of the LLVM package here: [LLVM Downloads](http://releases.llvm.org/download.html)
+
+- To run clang-format on all staged files use the following command:
   ```
   git diff --cached --name-only | egrep '[.](cpp|h|mm)$' | xargs clang-format -i
   ```
+  - You will have to restage any modified files.
+  - This operation may produce `.TMP` backups of the original files that were modified. Make sure you do not include these files when restagining. You may move or delete them. To ignore them by git, copy this `*.TMP` into a new line in the `exclude` file in the `.git\info` directory inside your local repository. Note that `.git` is a hidden folder under Windows.
+  - Windows users: To delete all `.TMP` files in your repository, run command prompt from the root repository location with the following command `del /s /q *.tmp`
+  - Windows users: Unfortunately clang-format is only available for official download packed inside a rather large LLVM installer, whereas clang-format is very small. Thankfully there is a way to skip the installer. The LLVM installer executable can be opened by [7-Zip](http://www.7-zip.org/download.html) directly as if it's a ZIP file. Download the LLVM installer but don't run it. Rename the LLVM installer's `.exe` extension to `.7z` or drag&drop it onto `7zFM.exe` located in the 7-Zip root installation directory. Open the `bin` folder, locate `clang-format.exe` and place it in `usr\bin` subdirectory of your Git installation path.
```

Comment:
Remove this. It's not in the scope of this file to tell users how to do stupid things to the LLVM installer.

The above can probably be removed as well. Run correctly, clang-format won't leave .tmp files behind.

Reply:
Provide the "correctly" command then.

Reply:
It's currently in the contributing.md file. You're the only person having this problem.

Reply:
You assume I'm the only one, for all the future newcomers I guess, and you can't tell me what is the correct way. So as long as you know and you have no problems, this doesn't need to be updated.

I didn't thought these files are meant for the people who are already contributing..